src/hal/drivers/hal_bb_gpio.c

```diff
@@ -78,10 +78,33 @@ void configure_control_module() {
 }
 
 void configure_gpio_port(int n) {
+    volatile void *cm_per;  // pointer to clock manager registers
+    volatile unsigned int *regptr;
+    unsigned int regvalue;
+
     int fd = rtapi_open_as_root("/dev/mem", O_RDWR);
 
     gpio_ports[n] = hal_malloc(sizeof(bb_gpio_port));
 
+    // need to verify that port is enabled and clocked before accessing it
+    // port 0 is always mapped, the others need checked
+    if ( n > 0 ) {
+	cm_per = mmap(0,CM_PER_LEN, PROT_READ | PROT_WRITE, MAP_SHARED, fd, CM_PER_ADDR);
+	if(cm_per == MAP_FAILED) {
+	    rtapi_print_msg(RTAPI_MSG_ERR, "%s: ERROR: Unable to map Clock Module: %s\n", modname, strerror(errno));
+	    exit(1);
```

Comment:
It looks like you were merely following the surrounding code style by calling exit here.  I don't know if this is OK in MK, but in linuxcnc it isn't; calling exit() here would make the whole rtapi_app process exit.  looks like it'll be just about as easy to take this patch first and then go back and fix it, though.